### PR TITLE
Radio test support

### PR DESCRIPTION
**Pull request by ajayparida**
Changes for supporting radio-test mode.

### CODE REVIEW

File: src/debugfs/main.c [hunk 1]
```diff
@@ -42,18 +42,20 @@ int nrf_wifi_wlan_fmac_dbgfs_init(struct nrf_wifi_ctx_lnx *rpu_ctx_lnx)
 	if (!rpu_ctx_lnx->dbgfs_wlan_root)
 		goto out;
 
+
```
Comment (krish2718) [inline src/debugfs/main.c:45]: unnecessary newline

File: src/debugfs/wlan_fmac_conf.c [hunk 2]
```diff
@@ -221,20 +2255,20 @@ static const struct file_operations fops_wlan_conf = {
 };
 
 int nrf_wifi_wlan_fmac_dbgfs_conf_init(struct dentry *root,
-				       struct nrf_wifi_ctx_lnx *rpu_ctx_lnx)
```
Comment (krish2718) [inline src/debugfs/wlan_fmac_conf.c:224]: can you please move these renames to a separate commit?

**Review by krish2718 — CHANGES_REQUESTED** [review]
Please don't use `changes` in the commit log, every commit is changing something, the commit log should describe what and why are the changes.

File: Makefile [hunk 3]
```diff
@@ -8,8 +8,8 @@ DTS = dts/nrf70_rpi_interposer.dts
 DRV_FUNC_NAME =  _wifi_fmac
 
 # If mode != STA throw error
-ifneq ($(MODE), STA)
-$(error Invalid MODE=$(MODE) specified. Valid values are STA)
+ifneq (, $(filter STA, RADIO-TEST, $(MODE)))
```
[the remark this answers is not in the document]
Comment (krish2718) [reply Makefile:11]: You also need to update CI as well

File: src/debugfs/main.c [hunk 4]
```diff
@@ -42,18 +42,20 @@ int nrf_wifi_wlan_fmac_dbgfs_init(struct nrf_wifi_ctx_lnx *rpu_ctx_lnx)
...
 		goto out;
 
+#ifndef CONFIG_NRF700X_RADIO_TEST
 	status = nrf_wifi_wlan_fmac_dbgfs_twt_init(rpu_ctx_lnx->dbgfs_wlan_root,
 						   rpu_ctx_lnx);
 
 	if (status != NRF_WIFI_STATUS_SUCCESS)
 		goto out;
-
+#endif
```
[the remark this answers is not in the document]
Comment (ajayparida) [reply src/debugfs/main.c:57]: Hi I din't get this. Please check  if it has been fixed.

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (ajayparida) [reply Makefile]: Makefile chnages updated,

[the remark this answers is not in the document]
Comment (krish2718) [reply Makefile]: I think you mean CI updated, but the README is still not updated.

**Review by krish2718 — COMMENTED** [review]
can you please rename 

`wlan_fmac_conf.c` -> `radio_test.c` (if there are common functions we can move it to main.c)
wlan_fmac_stats.c` -> `stats.c`

File: README.md [hunk 5]
```diff
@@ -192,6 +193,83 @@ Ensure that the `nrf_wifi_fmac_sta.ko` and `dts/nrf70_rpi_interposer.dtbo` files
...
+    ```bash
+    sudo insmod nrf_wifi_fmac_radio_test.ko
+    ```
+3. The commands used for radio test can be found [`here`](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/wifi/radio_test/radio_test_subcommands).
+4. To set different test parameters `debugfs` is used
+   
+   > For EK to transmit 10000 packets (TX transmit count) with the required modulation, TX power and channel
+     (e.g. 11g, 54 Mbps, channel 11):
+   ```bash
+   #echo init=11 > /sys/kernel/debug/nrf/wifi/conf
```
Comment (krish2718) [inline README.md:216]: space after `#`

File: README.md [hunk 6]
```diff
@@ -192,6 +193,83 @@ Ensure that the `nrf_wifi_fmac_sta.ko` and `dts/nrf70_rpi_interposer.dtbo` files
...
+   #echo tx_pkt_tput_mode=0 > /sys/kernel/debug/nrf/wifi/conf
+   #echo tx_pkt_rate=54 > /sys/kernel/debug/nrf/wifi/conf
+   #echo tx_pkt_len=1024 > /sys/kernel/debug/nrf/wifi/conf
+   #echo tx_pkt_gap=100 > /sys/kernel/debug/nrf/wifi/conf
+   #echo tx_pkt_num=10000 > /sys/kernel/debug/nrf/wifi/conf
+   #echo tx=1 > /sys/kernel/debug/nrf/wifi/conf
+   ```
+   Once set check the configuration settings in `/sys/kernel/debug/nrf/wifi/conf`. Typically the configuration will be as below
+   ```bash
+   root@ubuntu:~# cat /sys/kernel/debug/nrf/wifi/conf
```
Comment (krish2718) [inline README.md:226]: can we remove the `root@ubuntu`

**Review by krish2718 — COMMENTED** [review]
Can you please add commit log to 3rd commit?

**Comment by ajayparida** [comment]
> can you please rename
> 
> `wlan_fmac_conf.c` -> `radio_test.c` (if there are common functions we can move it to main.c) wlan_fmac_stats.c`->`stats.c`

Renamed files as suggested above.

**Review by krish2718 — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -192,6 +193,83 @@ Ensure that the `nrf_wifi_fmac_sta.ko` and `dts/nrf70_rpi_interposer.dtbo` files
...
+    ```
+   Ensure that the `rf_wifi_fmac_radio_test.ko` and `dts/nrf70_rpi_interposer.dtbo` files are generated.
+2. Load the driver
+    ```bash
+    sudo insmod nrf_wifi_fmac_radio_test.ko
+    ```
+3. The commands used for radio test can be found [`here`](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/wifi/radio_test/radio_test_subcommands.html).
+4. To set different test parameters `debugfs` is used
+   
+   > For EK to transmit 10000 packets (TX transmit count) with the required modulation, TX power and channel
```
Comment (krish2718) [inline README.md:213]:
```suggestion
   > For nRF70 series device to transmit 10000 packets (TX transmit count) with the required modulation, TX power and channel
```

File: src/debugfs/radio_test.c [hunk 8]
```diff
@@ -0,0 +1,1996 @@
...
+
+static int nrf_wifi_wlan_fmac_conf_disp(struct seq_file *m, void *v)
+{
+	struct rpu_conf_params *conf_params = NULL;
+
+	conf_params = &ctx->conf_params;
+
+	seq_puts(m, "************* Configured Parameters ***********\n");
+
+#ifndef CONFIG_NRF700X_RADIO_TEST
```
Comment (krish2718) [inline src/debugfs/radio_test.c:72]: Is this needed? This file should only have radio test related code.

File: src/debugfs/stats.c [hunk 9]
```diff
@@ -509,8 +509,13 @@ static int nrf_wifi_wlan_fmac_dbgfs_stats_show(struct seq_file *m, void *v)
...
 	int op_mode = RPU_OP_MODE_MAX;
 	struct rpu_op_stats *stats = NULL;
-
+#ifdef CONFIG_NRF700X_RADIO_TEST
+	unsigned int i = 0;
+#endif
 	rpu_ctx_lnx = (struct nrf_wifi_ctx_lnx *)m->private;
+#ifdef CONFIG_NRF700X_RADIO_TEST
+	op_mode = rpu_ctx_lnx->conf_params.op_mode,
+#endif
```
Comment (krish2718) [inline src/debugfs/stats.c:518]: In general we are following closing guards in comments

File: README.md (hunk 7 above)
Comment (ajayparida) [reply README.md]: Update as per suggestion

File: src/debugfs/stats.c (hunk 9 above)
Comment (ajayparida) [reply src/debugfs/stats.c:518]: This is no more needed. Separated radio test and STA mode functionalities as per below comment.

File: src/debugfs/radio_test.c (hunk 8 above)
Comment (ajayparida) [reply src/debugfs/radio_test.c]: Separated radio test and STA mode functionalities in seaparate file.

File: src/debugfs/stats.c (hunk 9 above)
Comment (krish2718) [reply src/debugfs/stats.c:518]: I was referring to below but we can do that later.

`#endif /* CONFIG_NRF700X_RADIO_TEST */`
